libraries/apollo-mockserver/src/commonMain/kotlin/com/apollographql/apollo3/mockserver/WebSocketMockServer.kt

```diff
@@ -0,0 +1,39 @@
+package com.apollographql.apollo3.mockserver
+
+import com.apollographql.apollo3.annotations.ApolloExperimental
+import kotlinx.coroutines.flow.Flow
+import kotlin.random.Random
+
+@ApolloExperimental
+interface WebSocketMockServer {
+  @ApolloExperimental
+  sealed class WebSocketEvent {
+    @ApolloExperimental
+    class Connect(val sessionId: String, val headers: Map<String, String>) : WebSocketEvent()
+
+    @ApolloExperimental
+    class TextFrame(val sessionId: String, val text: String) : WebSocketEvent()
```

Comment:
Naming bikeshedding: I thought a Text message was split accross multiple frames and therefore this would need to be called `TextMessage` (or just `Text`)? Similarly, `WebSocketEvent` could be named `WebSocketMessage`?

Reply:
In `events` however you can receive "frames" (which I agree it could make sense to rename "messages"), but also other things that are not really messages  (connect, close, error) IMO.

Reply:
Gotcha 👍 

I was expecting an API similar to MockWebserver where we don't have anything like `Connect` or `Error`. All we can get is the HTTP request. 

Similarly for WebSockets, I would expect to be able to record the initial HTTP handshake request as well as subsequent Close/Ping/Pong control frames (or messages because these messages are always a single frame) and data messages (not individual frames).

Just my 2 cents, feel free to discard.